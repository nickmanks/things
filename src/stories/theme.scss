
$storybook-container-shadow: 1px 1px 5px 1px rgba(0, 0, 0, .37);

.story-container {
  margin: 3rem;
  box-shadow: $storybook-container-shadow;
  width: 100%;
  height: 65rem;
  border-radius: .4rem;

  > div {
    height: 100%;
    width: 100%;

    > div:first-child {
      height: 100%;
      width: 100%;
    }
  }
}
