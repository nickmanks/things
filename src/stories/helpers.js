import React from 'react';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../reducers';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'shards-ui/dist/css/shards.min.css';
import './theme.scss';


export function withCustomContainer(backgroundColor) {
  const CustomContainer = (story)=> (
    <div className={'story-container'} style={{backgroundColor}}>
      {story()}
    </div>
  );
  CustomContainer.displayName = 'CustomContainer';
  return CustomContainer;
}

export const storyStore =
  (state)=> createStore(reducers, state, applyMiddleware(thunk));
