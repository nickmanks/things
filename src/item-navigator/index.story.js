/* global module */
import React from 'react';
import {Provider} from 'react-redux';
import {storiesOf} from '@storybook/react';
import {storyStore} from '../stories/helpers';
import ItemNavigator from '.';
import testItems from '../testing/test-items';


storiesOf('Connected/ItemNavigator', module)
  .add('with multiple items', ()=> {
    const store = storyStore({
      things: {items: testItems}
    });

    return (
      <Provider store={store}>
        <ItemNavigator store={store} />
      </Provider>
    );
  })
  .add('with no items', ()=> {
    const store = storyStore({
      things: {items: {}}
    });

    return (
      <Provider store={store}>
        <ItemNavigator store={store} />
      </Provider>
    );
  });
