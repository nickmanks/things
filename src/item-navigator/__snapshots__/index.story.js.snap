// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Connected/ItemNavigator with multiple items 1`] = `
<ItemNavigator store={{}} items={[{"id":"test-id-1","name":"Today do things","category":"thing","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":"12m","status":"done"},{"id":"test-id-2","name":"Thing is due","category":"due","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":"2d","status":"progress"},{"id":"test-id-3","name":"Create a thing","category":"create","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":"3h","status":"ready"},{"id":"test-id-4","name":"Create a thing","category":"create","description":"dadasdasd.","created":"3h","status":"done"}]} selected={null} onCloseModal={"function"} onDescriptionChange={"function"} onCategoryChange={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fragment>
    <Container tag={"div"}>
      <div className={"container"}>
        <Row tag={"div"} key={"0"}>
          <div className={"row"}>
            <Col tag={"div"} breakpoints={["xs","sm","md","lg","xl"]}>
              <div className={"col"}>
                <Connect(Item) item={{"id":"test-id-1","name":"Today do things","category":"thing","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":"12m","status":"done"}} />
              </div>
            </Col>
            <Col tag={"div"} breakpoints={["xs","sm","md","lg","xl"]}>
              <div className={"col"}>
                <Connect(Item) item={{"id":"test-id-2","name":"Thing is due","category":"due","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":"2d","status":"progress"}} />
              </div>
            </Col>
            <Col tag={"div"} breakpoints={["xs","sm","md","lg","xl"]}>
              <div className={"col"}>
                <Connect(Item) item={{"id":"test-id-3","name":"Create a thing","category":"create","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":"3h","status":"ready"}} />
              </div>
            </Col>
            <Col tag={"div"} breakpoints={["xs","sm","md","lg","xl"]}>
              <div className={"col"}>
                <Connect(Item) item={{"id":"test-id-4","name":"Create a thing","category":"create","description":"dadasdasd.","created":"3h","status":"done"}} />
              </div>
            </Col>
          </div>
        </Row>
        <Row tag={"div"} key={"1"}>
          <div className={"row"}>
            <Col tag={"div"} breakpoints={["xs","sm","md","lg","xl"]}>
              <div className={"col"}>
                <Connect(OutlineItem) />
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </Fragment>
</ItemNavigator>
`;

exports[`Storyshots Connected/ItemNavigator with no items 1`] = `
<ItemNavigator store={{}} items={[]} selected={null} onCloseModal={"function"} onDescriptionChange={"function"} onCategoryChange={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fragment>
    <Container tag={"div"}>
      <div className={"container"}>
        <Row tag={"div"} key={"alone"}>
          <div className={"row"}>
            <Col tag={"div"} breakpoints={["xs","sm","md","lg","xl"]}>
              <div className={"col"}>
                <Connect(OutlineItem) />
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </Fragment>
</ItemNavigator>
`;
